core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -343,38 +331,35 @@ public ParsedQuery parse(XContentParser parser) {
      */
     @Nullable
     public ParsedQuery parseInnerFilter(XContentParser parser) throws IOException {
```

Comment:
I agree, unfortunately these are used all over the place. I will have another look but again maybe that is better suited for a follow up PR.